core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java

```diff
@@ -106,11 +119,11 @@ public ZooLock(AccumuloConfiguration conf, String path, UUID uuid) {
     }
   }
 
-  protected ZooLock(ZooKeeper zookeeper, String path, UUID uuid) {
+  protected ZooLock(ZooKeeper zookeeper, ZooLockPath path, UUID uuid) {
```

Comment:
Is line 177 correct, without IDE help it looks like path must be a String there.

Reply:
This line? You mention 177 but I see the method header in your snipit. 

https://github.com/apache/accumulo/blob/cac03f54971c2d2ebcfffd03ed593cf6a922b354/core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java#L177

Reply:
Yes - had issues getting GH to allow me to comment in the correct spot.

Reply:
Interesting. I see what you mean but I haven't noticed any issue with it and IntelliJ does not show an error either. I can add the `toString()` call for sanity reasons if you think that is best.

Reply:
If the `toString()` is correct, then that line is correct. Java automatically calls `toString()` on non-null objects when doing string concatenation (and uses literal "null" if null).

Reply:
This is one reason that I prefer an explicit getPath() - if for any reason toString() was changed, it may be harder than it needs to be to find places where the behavior of toString value is being used, esp with an automatic conversion. But again, my opinion...

Reply:
@EdColeman In general, I agree with you, on the need for explicit methods for serialization, rather than relying on `toString()`, but in this case, I don't think it makes sense, since these are merely more strongly typed strings, so `toString()` should always be implemented correctly. It's more intuitive to think of these as more strongly typed strings, than it is to consider them path object types that have getters to a contained path object type.